### PR TITLE
Add more args to SFT example

### PR DESCRIPTION
**Pull request by photomz**
The SFT example `sft_trainer.py` does not support all `HFArguments` available. This PR adds argparser flags for common `HFArguments`, so that the example is more usable.

This is a very simple feature change.

### CODE REVIEW

File: examples/scripts/sft_trainer.py [hunk 1]
```diff
@@ -99,6 +103,10 @@ class ScriptArguments:
     num_train_epochs=script_args.num_train_epochs,
     max_steps=script_args.max_steps,
     report_to=script_args.log_with,
+    save_steps=script_args.save_steps,
+    save_total_limit=script_args.save_total_limit,
+    push_to_hub=script_args.push_to_hub,
```
[the remark this answers is not in the document]
Comment (photomz) [reply examples/scripts/sft_trainer.py:108]: Yes, I trained `llama2-7b` and successfully pushed a private model.

How should I test this more rigorously @lvwerra ?

[the remark this answers is not in the document]
Comment (lvwerra) [reply examples/scripts/sft_trainer.py:108]: Does this push the base model weights or the adapters or both?

**Comment by lvwerra** [comment]
Could your run `make precommit` to fix the style issues?

File: examples/scripts/sft_trainer.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (photomz) [reply examples/scripts/sft_trainer.py:108]: If PeFT/LoRA, then adapters; otherwise the tuned base model.

**Comment by photomz** [comment]
@lvwerra I've fixed them, tests should pass now?

**Review by younesbelkada — APPROVED** [review]
Thanks!

**Pull request merged**